includes/acl/compression/impl/optimize_looping.h

```diff
@@ -95,6 +95,7 @@ namespace acl
 			convert_transforms_args_local.num_dirty_transforms = 2;
 			convert_transforms_args_local.transforms = &local_transforms[0];
 			convert_transforms_args_local.num_transforms = 2;
+			convert_transforms_args_local.sample_index = last_sample_index;
```

Comment:
I think we may want to move this below on line 144. There we use this to convert both the first/last samples together but perhaps we need to split it over two calls so that we can set this

Reply:
I saw that, but I think it is actually fine (which I tried to hint at with the commit message). When you think about this from a higher level, what you are trying to do is figure out if the first sample is a suitable replacement for the last sample. Therefore, you need to subject the first sample to the last sample's external influences, because those are the influences that would be taking place when sampling the end of the clip within the one-sample region of optimization.

If you replace "reference" with "some_lossy_value from_a_genie", it starts to be more obvious why the sample index must be the last for both samples.

Reply:
That makes sense. I think we'll need to document this however, as it isn't immediately obvious by looking at the code but you're right.